File: source/O6.jl
# This is the Julia version of the Open Ended D6 rolls used in the Amar RPG
# See http://d6gaming.org for details about the Amar Role-Playing Game system


function D(x)
    rand(1:x)
end

m = true; mark = ""; d = D(6); t = D(6)
if d == 1
	while t < 4
		if t == 1 && !m; m = true
		elseif t == 1 && m; mark = " fumble"
		else; m = false
		end
		d -= 1
		t = D(6)
	end
end
if d == 6
	while t > 3
		if t == 6 && !m; m = true
		elseif t == 6 && m; mark = " critical"
		else; m = false
		end
		d += 1
		t = D(6)
	end
end
println("$d $mark")

File: source/O6.rb
# This is the Ruby version of the Open Ended D6 rolls used in the Amar RPG
# See http://d6gaming.org for details about the Amar Role-Playing Game system

def d6
	return rand(1..6)
end

def o6
	m = true ; mark = "" ; t = d6
	d = d6

	if d == 1
		while t < 4
			d -= 1
			t == 1 ? m == true ? mark = " fumble" : m = true : m = false
			t = d6
		end
	end

	if d == 6
		while t > 3
			d += 1
			t == 6 ? m == true ? mark = " critical" : m = true : m = false
			t = d6
		end
	end

	puts "#{d.to_s}#{mark}"
end

o6
